{"blogid": "123512684", "writerAge": "码龄2年", "writerBlogNum": "44", "writerCollect": "49", "writerComment": "73", "writerFan": "14", "writerGrade": "3级", "writerIntegral": "607", "writerName": "荒-漠", "writerProfileAdress": "writer_image\\profile_123512684.jpg", "writerRankTotal": "40177", "writerRankWeekly": "123366", "writerThumb": "44", "writerVisitNum": "52294", "blog_read_count": "3865", "blog_time": "已于 2022-04-06 15:29:42 修改", "blog_title": "Tp5在Linux部署报错During inheritance of JsonSerializable: Uncaught think\\exception\\ErrorException", "content": "<div class=\"article_content clearfix\" id=\"article_content\">\n<link href=\"style.css\" rel=\"stylesheet\"/>\n<div class=\"htmledit_views\" id=\"content_views\">\n<p>问题描述：tp项目在本地调试没有任何问题，在linux第一次调试也莫有问题；后来xampp升级忘记备份了 就出现了这个问题。 </p>\n<p></p>\n<p><img alt=\"\" height=\"902\" src=\"image\\86a1e561d3c540968801a3490822d59e.png\" width=\"1200\"/></p>\n<p>报错信息翻译后：</p>\n<p>在JsonSerializable的继承过程中：<br/> 未捕获think\\exception\\ErrorException:返回类型属于think\\Model::jsonSerialize()两者都应该与JsonSerializable::jsonSerialize():混合的，<br/> 或应使用【返回类型将改变】属性应用于临时抑制中的通知 </p>\n<p>发现这个报错信息也不能基于全部，以后还是要减少对debug的依赖。</p>\n<p>尝试了修改use think\\Exception 改为use Exception 和替换整体think文件 更新apache 在nginx测试都莫有效果 <span style=\"color:#fe2c24;\">还是要坚信think包不会有问题的 有问题的还是自己的代码</span></p>\n<p></p>\n<p>最后没有办法了，开始从接口一步步走 发现在调取系统Model返回的类型不正确；尝试<span style=\"color:#fe2c24;\"><strong>修改了一下use think\\Model 换成了 use think\\console\\command\\make\\Model 成功解决了这个问题</strong></span>；就是在引入的时候没有引入成功。在windows上莫有任何问题 在linux这方面是比较严格。</p>\n<p></p>\n<p><img alt=\"\" height=\"682\" src=\"image\\efc1e33a0aef4d9abe9d4649dfdb36e9.png\" width=\"1140\"/></p>\n<hr/>\n<h2><span style=\"color:#fe2c24;\"><strong>2022.4.6在次遇到这种类似问题 发现最终问题是php的版本太高8.1以上</strong></span></h2>\n<p>解决办法:<a class=\"has-card\" href=\"https://blog.csdn.net/qq5201314wx/article/details/123990897\" title=\"During inheritance of Iterator: Uncaught think\\exception\\ErrorException: Return type of_荒-漠的博客-CSDN博客\"><span class=\"link-card-box\"><span class=\"link-title\">During inheritance of Iterator: Uncaught think\\exception\\ErrorException: Return type of_荒-漠的博客-CSDN博客</span><span class=\"link-desc\">tp5项目在本地运行没有问题,在linux上报错其实不是linux的问题 ,最后发现是本地和线上linux部署的环境不同才导致这种问题tp5在php版本8.1以上会出现好多不兼容的问题 ,最好的解决办法就是给php进行降级操作During inheritance of Iterator: Uncaught think\\exception\\ErrorException: Return type of PhpOffice\\PhpSpreadsheet\\Worksheet\\Iterator::val.</span><span class=\"link-link\"><img alt=\"\" class=\"link-link-icon\" src=\"https://g.csdnimg.cn/static/logo/favicon32.ico\"/>https://blog.csdn.net/qq5201314wx/article/details/123990897</span></span></a></p>\n<p></p>\n<p></p>\n</div>\n</div>"}